rfc/core language/4956-cluster-pools.md

```diff
@@ -0,0 +1,100 @@
+# [RFC] Select cluster execution when triggering an execution
+
+**Authors:**
+
+- @RRap0so
+- @andresgomezfrr
+
+## 1 Executive Summary
+
+When triggering a Flyte LaunchPlan, Workflow or Task, Flyte permits the user to define a ClusterPool from a predefined list of pools that are set in the FlyteAdmin configuration. For example with `flytectl create execution
+--clusterPool string` 
+
+This provides greater flexibility compared to the existing feature where an [Execution Cluster Label](https://docs.flyte.org/en/latest/deployment/deployment/multicluster.html#configure-execution-cluster-labels) can be set at the Project/Domain level.
+
+## 2 Motivation
+
+The primary motivation for this proposal is to select the clusters where an execution is to be performed dynamically at execution time, without the constant need to overwrite the Execution Cluster Label.
+
+Additionally, we would like to have the possibility of different executions targeting different clusters, even if they are published under the same Project/Domain. This is not entirely feasible using the current Execution Cluster Label.
+
+
+## 3 Proposed Implementation
+
+We plan to use the `cluster_assignement` field (refer to [proto](https://github.com/flyteorg/flyte/blob/a2a63eb893bdff3c2d1a01242d0d25b73f287b1f/flyteidl/protos/flyteidl/admin/execution.proto#L315)) in each execution, this is a ClusterAssignment object with a `cluster_pool_name` which would be the field that contains a string with the corresponding ClusterPool ID.
+
+In the FlyteAdmin configuration we have a list of `clusterpool_config` with all the different available pools and corresponding clusters.
+
+**Note**: Today there are configurations to assign domains to specific pools, see [PR](https://github.com/flyteorg/flyte/pull/4208) and `cluster_pool` [interface](https://github.com/flyteorg/flyte/blob/962d45827646f106265630a0fb1d9ae74968bc32/flyteadmin/pkg/runtime/interfaces/cluster_pools.go). We propose another configuration to assign Clusters to a clusterPoolName by having another map under cluster object with weights and cluster assignments.
+
+Proposed configuration:
+```
+    clusters:
```

Comment:
As said in the contributors' sync:

Love the proposal, my only wish would be to figure out whether there is a way to simplify this config here by not having to configure a cluster twice, once for the existing project/domain mechanism, once for the new mechanism. Would be nice if the cluster configuration could be shared.

Reply:
@fg91 I think it should be possible by expanding the `ExecutionParameters` struct. By adding the `ExecutionClusterLabel` to this [struct](https://github.com/flyteorg/flyte/blob/a2a63eb893bdff3c2d1a01242d0d25b73f287b1f/flyteadmin/pkg/workflowengine/interfaces/executor.go#L20) and making sure we pass it when executing it should make things simpler and avoid having double configs.

This will mean a bit of re-writing, since it's late I'll also cross-check my thinking and get back with some certainties next week.